components/tidb_query_datatype/src/codec/table.rs

```diff
@@ -30,6 +30,59 @@ pub const TABLE_PREFIX_KEY_LEN: usize = TABLE_PREFIX_LEN + ID_LEN;
 // the maximum len of the old encoding of index value.
 pub const MAX_OLD_ENCODED_VALUE_LEN: usize = 9;
 
+#[derive(Debug)]
+pub enum Handle<'a> {
+    Int(i64),
+    Common(Vec<BytesSlice<'a>>),
+}
+
+impl<'a> Handle<'a> {
+    pub fn into_i64(self) -> Result<i64> {
+        match self {
+            Handle::Int(i) => Ok(i),
+            _ => Err(invalid_type!(
+                "Expect an integer handle, got a common handle"
+            )),
+        }
+    }
+
+    pub fn into_common(self) -> Result<Vec<BytesSlice<'a>>> {
+        match self {
+            Handle::Common(slices) => Ok(slices),
+            _ => Err(invalid_type!(
+                "Expect a common handle, got an integer handle"
+            )),
+        }
+    }
+}
+
+impl<'a> TryFrom<BytesSlice<'a>> for Handle<'a> {
+    type Error = Error;
+
+    fn try_from(mut key: BytesSlice<'a>) -> Result<Self> {
+        use std::cmp::Ordering;
+        // Expects a record key.
+        check_record_key(key)?;
+
+        match key.len().cmp(&RECORD_ROW_KEY_LEN) {
+            Ordering::Less => Err(invalid_type!("record key expected, but got {:?}", key)),
+            Ordering::Equal => Ok(Handle::Int(
```

Comment:
I'm worried about the correctness of this method. When it is a cluster index handle, the length can be `RECORD_ROW_KEY_LEN` as well.

Reply:
Clustered index length is always longer than RECORD_ROW_KEY_LEN.